projects/find/find.py

```diff
@@ -0,0 +1,57 @@
+import os
+import sys
+
+"""this method is created to find all file and directory, given a specific path. Optionally, your can use -name flag
+to search only the files that match to a specific name"""
+
+
+def find(path, option=None, filename=None):
+    """this method uses a bfs approach to resolve to the problem of finding all files and directories."""
+    if not os.path.exists(path):
+        return "Path non corretto o inestitente: si prega di riprovare"
+    queue = []
+    queue.append(path)
+    # marked = [False] * len(list)
+    # edge_to = [None] * len(list)
+    list = []
+    if filename is None and option is None:
+        while len(queue) > 0:
+            queue_element = queue.pop(0)
+            adj_list = os.listdir(queue_element)
+            if adj_list is not []:
+                for adj in adj_list:
+                    if os.path.isfile(queue_element + adj):
```

Comment:
for paths, you should use `os.path.join(queue_element, adj)`. that will sort out the `/` for you and you won't have to add them manually